debezium-server/debezium-server-kafka/src/main/java/io/debezium/server/kafka/KafkaChangeConsumer.java

```diff
@@ -0,0 +1,108 @@
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.DebeziumException;
+import io.debezium.engine.ChangeEvent;
+import io.debezium.engine.DebeziumEngine;
+import io.debezium.engine.DebeziumEngine.RecordCommitter;
+import io.debezium.server.BaseChangeConsumer;
+import io.debezium.server.CustomConsumerBuilder;
+
+/**
+ * An implementation of the {@link DebeziumEngine.ChangeConsumer} interface that publishes change event messages to Kafka.
+ */
+@Named("kafka")
+@Dependent
+public class KafkaChangeConsumer extends BaseChangeConsumer implements DebeziumEngine.ChangeConsumer<ChangeEvent<Object, Object>> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaChangeConsumer.class);
+
+    private static final String PROP_PREFIX = "debezium.sink.kafka.";
+    private static final String PROP_PREFIX_PRODUCER = PROP_PREFIX + "producer.";
+
+    private KafkaProducer<Object, Object> producer;
+
+    @Inject
+    @CustomConsumerBuilder
+    Instance<KafkaProducer<Object, Object>> customKafkaProducer;
+
+    @PostConstruct
+    void start() {
+        if (customKafkaProducer.isResolvable()) {
+            producer = customKafkaProducer.get();
+            LOGGER.info("Obtained custom configured KafkaProducer '{}'", producer);
+            return;
+        }
+
+        final Config config = ConfigProvider.getConfig();
+        producer = new KafkaProducer<>(getConfigSubset(config, PROP_PREFIX_PRODUCER));
+        LOGGER.info("consumer started...");
+    }
+
+    @PreDestroy
+    void stop() {
+        LOGGER.info("consumer destroyed...");
+        if (producer != null) {
+            try {
+                producer.close(Duration.ofSeconds(5));
+            }
+            catch (Throwable t) {
+                LOGGER.warn("Could not close producer", t);
+            }
+        }
+    }
+
+    @Override
+    public void handleBatch(final List<ChangeEvent<Object, Object>> records, final RecordCommitter<ChangeEvent<Object, Object>> committer) throws InterruptedException {
+        final List<Future<RecordMetadata>> futures = new ArrayList<>();
+        for (ChangeEvent<Object, Object> record : records) {
+            try {
+                // TODO: change log level to trace
+                LOGGER.info("Received event '{}'", record);
+                futures.add(producer.send(new ProducerRecord<>(record.destination(), record.key(), record.value())));
+                committer.markProcessed(record);
+            }
+            catch (Exception e) {
+                throw new DebeziumException(e);
+            }
+        }
+
+        final List<Long> offsets = new ArrayList<>();
```

Comment:
Is this only for logging? If yes then please all the protion of the code shoud be graded with `LOGGER.isTraceEnabled()`

Reply:
I also used it as a way to `mark that the batch of records have been processed`. so yes, its not only for logging purposes. Or should we mark the batch as processed even though the batch may not have been sent to kafka?

Reply:
It seems to me in the code that the variable is really used only in the log message.

Reply:
if you're referring only to the list of offsets variable, yes that's true. But since getting the offsets will wait for the message to be committed in kafka, the last line `committer.markBatchFinished()` will IMO be executed only if all records are committed in kafka.  

If i would surround this in a logging enabled block and trace logging is not enabled, then a call to markBatchFinished() will not wait for all records to be committed in kafka.

But i will  update the patch as suggested

Reply:
@Alfusainey I see now. I am sorry for the confusion as I mixed Connect offsets and kafka offsets together. So yes, it really makes sense to wait till the write is completed!
Then I'd still have a different proposal. Maybe you could use the send with callback method and use lath or barrier and wail for all to be completed instead of iteration over the list?

Reply:
> Then I'd still have a different proposal. Maybe you could use the send with callback method and use lath or barrier and wail for all to be completed instead of iteration over the list?

yes, that make sense! ok, I will get back to this PR this weekend. The deadline for submitting the GSoC proposal is tomorrow and i have another deadline this Friday 😄